classes/frontend.php

```diff
@@ -62,7 +64,13 @@ protected function get_javascript_init_params($course, cm_info $cm = null,
             $sql2 = "SELECT * FROM {course}
                     ORDER BY fullname ASC";
             $other = $DB->get_records_sql($sql2);
-            //$other = get_courses();
+
+            // Filter courses for access, users should only be able to create restrictions on courses they can edit.
+            $other = array_filter($other, function($course) {
```

Comment:
Hi Peter. The logic here is sound. 

Though it potentially is querying the database and doing a bunch of extra things, when in the following loop, not all entries might be included. Unsure if there's an appetite for this to be optimised a bit further, as this mainly effects people who can edit (and not students I believe).

Thoughts?